src/script/cpatch.py

```diff
@@ -511,79 +528,37 @@ def _copy_binary(self, src_path, dst_path):
             pass
         os.link(src_path, dst_path)
 
-    def _bins_and_libs(self, prefix, bin_patterns, lib_patterns):
+    def _bins_and_libs(self, exclude_prefixes, bin_patterns, lib_patterns):
         out = []
 
         bin_src = self._ctx.build_dir / "bin"
-        bin_dst = self._workdir / f"{prefix}_bin"
+        bin_dst = self._workdir / "tmp_bin"
         bin_dst.mkdir(parents=True, exist_ok=True)
         for path in bin_src.iterdir():
             if any(path.match(m) for m in bin_patterns):
+                if any(path.match(f"{m}*") for m in exclude_prefixes):
+                    continue
                 self._copy_binary(path, bin_dst / path.name)
-        out.append(f"ADD {prefix}_bin /usr/bin")
 
         lib_src = self._ctx.build_dir / "lib"
-        lib_dst = self._workdir / f"{prefix}_lib"
+        lib_dst = self._workdir / "tmp_lib"
         lib_dst.mkdir(parents=True, exist_ok=True)
         for path in lib_src.iterdir():
             if any(path.match(m) for m in lib_patterns):
+                if any(path.match(f"{m}*") for m in exclude_prefixes):
+                    continue
                 self._copy_binary(path, lib_dst / path.name)
-        out.append(f"ADD {prefix}_lib /usr/lib64")
 
         return out
 
-    def _conditional_libs(self, src_dir, name, destination, lib_patterns):
-        lib_src = self._ctx.build_dir / src_dir
-        lib_dst = self._workdir / name
-        lib_dst.mkdir(parents=True, exist_ok=True)
-        try:
-            for path in lib_src.iterdir():
-                if any(path.match(m) for m in lib_patterns):
-                    self._copy_binary(path, lib_dst / path.name)
-        except FileNotFoundError as err:
-            log.warning("skipping lib %s: %s", name, err)
-        return f"ADD {name} {destination}"
-
-    def _py_site_packages(self):
```

Comment:
I'm utterly confused by why you're removing steps, esp. ones related to python.

Reply:
latest commit restores _py_site_packages()